portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -154,4 +166,35 @@ private int getNumQuoteDisplayed(HttpServletRequest request, int defaultNumToDis
       return defaultNumToDisplay; 
     }
   }
+
+  /**
+   * Extracts the attributes of a quoteEntity and puts them into a new
+   * instance of Quote. 
+   */ 
+  private Quote extractQuoteFromEntity(Entity quoteEntity) {
+    long id = quoteEntity.getKey().getId();
+    String text = (String) quoteEntity.getProperty(TEXT);
+    long timestampMillis = (long) quoteEntity.getProperty(TIMESTAMP);
+    String userEmail = (String) quoteEntity.getProperty(USEREMAIL); 
+    String nickname = (String) quoteEntity.getProperty(NICKNAME); 
+    Quote newQuote = new Quote(id, text, timestampMillis, userEmail, nickname);
+    return newQuote; 
+  } 
+
+  /**
+   * Returns the nickname of the user with id, or empty String if the user has not set a nickname.
+   */
+  private String getUserNickname(String id) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
You'll probably want to use Strong consistency here. It might be nice to wrap this in a helper method so you can just call `getDatastoreService()` wherever you need it.

Reply:
Done